scripts/write_deployinst_input.py

```diff
@@ -217,63 +217,39 @@ def reactor_render(list, output_file, is_cyborg=False):
-                n_assem_batch=int(round(data['capacity']/3000 * 764)),
-                capacity=data['capacity'])
-        # if PHWR or CANDU,
-        elif data['type'].decode('utf-8') == 'PHWR' or 'CANDU' in data['type'].decode('utf-8'):
-            reactor_body = candu_template.render(
+
+        if data['type'].decode('utf-8') == spec_dict.keys():
+            # if the reactor type matches with the pre-defined dictionary,
+            # use the specifications in the dictionary.
+            reactor_str = data['type'].decode('utf-8')
+            reactor_body = spec_dict[reactor_str][0].render(
                 country=data['country'].decode('utf-8'),
-                type=data['type'].decode('utf-8'),
+                type=reactor_str,
                 reactor_name=name,
-                assem_size=int(80000/473),
-                n_assem_core=int(round(data['capacity']/500 * 473)),
-                n_assem_batch=60,
-                capacity=data['capacity'])
-        # if French PWR, use mox template for mox reactor
-        elif data['type'].decode('utf-8') == 'PWR' and data['country'].decode('utf-8') == 'France':
-            reactor_body = mox_reactor_template.render(country=data['country'].decode('utf-8'),
-                                                       reactor_name=name,
-                                                       type=data['type'].decode(
-                                                           'utf-8'),
-                                                       assem_size=446,
-                                                       n_assem_core=int(
-                round(data['capacity']/1000 * 193)),
-                n_assem_batch=int(
-                round(data['capacity']/3000 * 193)),
-                capacity=data['capacity'])
-        # if AP1000, use AP1000 specifcations
-        elif data['type'].decode('utf-8') == 'AP1000':
-            reactor_body = pwr_template.render(country=data['country'].decode('utf-8'),
-                                               reactor_name=name,
-                                               assem_size=467,
-                                               n_assem_core=157,
-                                               n_assem_batch=52,
-                                               capacity=data['capacity'])
-        # if EPR, use EPR specifcations
-        elif data['type'].decode('utf-8') == 'EPR':
-            reactor_body = pwr_template.render(country=data['country'].decode('utf-8'),
-                                               reactor_name=name,
-                                               assem_size=467,
-                                               n_assem_core=216,
-                                               n_assem_batch=72,
-                                               capacity=data['capacity'])
-        # if not any of the above, all go with PWR specification.
+                assem_size=spec_dict[reactor_str][1],
```

Comment:
the whole point of a dictionary is to be able to use meaningful keys rather than meaningless integers. See more informative structure suggestion above.

Reply:
I'm not quite sure what you mean. The key for `spec_dict` is the reactor types (BWR, PHWR, CANDU etc.)